File: Jessica/7.Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if x>2147483647 or x< -2147483648:
            return 0
        flag=0
        ret=0
        if x<0:
            flag=1
            x=abs(x)
        str1=str(x)
        
        if flag==1:
            ret=-1*int(str1[::-1])
        else:
            ret=int(str1[::-1])
        
        if ret>2147483647 or ret< -2147483648:
            return 0
        else:
            return ret

File: Jessica/7.Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {          
        int ret=0;
        while(x){
            int temp=x%10;
            if(ret>INT_MAX/10 || ret<INT_MIN/10)
                return 0;
            ret=ret*10+temp;
            x=x/10;            
        }
        return ret;
    }
    
};